pyopenms_viz/_matplotlib/core.py

```diff
@@ -660,39 +676,48 @@ def create_y_axis_plot(self, canvas=None):
         elif self.y_kind == "spectrum":
             y_plot_obj = self.get_vline_renderer(
                 data=y_data,
-                x=self.z,
+                x=self.z_original if "z_original" in y_data.columns else self.z,  # ✅ Use `z_original`
                 y=self.y,
                 by=self.by,
                 canvas=canvas,
                 config=self.y_plot_config,
             )
         else:
             raise ValueError(f"Invalid y_kind: {self.y_kind}")
-        ax = y_plot_obj.generate(None, None)
-
-        # self.plot_x_axis_line()
+        
+        ax = y_plot_obj.generate(None, None)  # ✅ Ensure we return `ax` since superclass does.
 
         ax.set_xlim((0, y_data[self.z].max() + y_data[self.z].max() * 0.1))
         ax.invert_xaxis()
         ax.set_title(None)
         ax.set_xlabel(self.y_plot_config.xlabel)
         ax.set_ylabel(self.y_plot_config.ylabel)
         ax.set_ylim(ax.get_ylim())
-        return ax
 
+        return ax  # ✅ Ensure return type consistency with superclass.
     def create_main_plot(self):
+        """
+        Implements PeakMap plotting for Matplotlib.
+        - Applies log scaling only to PeakMap colors.
+        - Uses raw intensities for marginal histograms.
+        - Keeps 3D PeakMap and annotation logic intact.
+        """
         if not self.plot_3d:
+            # ✅ Apply log scaling only to colors, not raw intensity values
+            log_intensity = np.log1p(self.data[self.z]) if self.z_log_scale else self.data[self.z]
+
             scatterPlot = self.get_scatter_renderer(
                 data=self.data,
                 x=self.x,
                 y=self.y,
-                z=self.z,
+                z=log_intensity,  # ✅ Only log-transform color scale
```

Comment:
I think z  is expected to be a string for the column variable, not a numpy array.